### ISSUE TITLE
Incorrect line and column numbers when a query is missing a semicolon

### ISSUE THREAD

**Issue by kevinburke1** [sqlc-dev/sqlc#263]
If you forget to end a query with a semicolon you get:

```sql
-- Example queries for sqlc
CREATE TABLE authors (
  id   BIGSERIAL PRIMARY KEY,
  name text      NOT NULL,
  bio  text
);

-- name: GetAuthor :one
SELECT * FROM authors
WHERE id = $1 LIMIT 1;

-- name: ListAuthors :many
SELECT * FROM authors
ORDER BY name

-- name: CreateAuthor :one
INSERT INTO authors (
  name, bio
) VALUES (
  $1, $2
)
RETURNING *;

-- name: DeleteAuthor :exec
DELETE FROM authors
WHERE id = $1;
```

```
# package querytest
query.sql:1:1: syntax error at or near "INSERT"
```

Which is confusing because I start looking at the next query instead of the one above it, and the `1:1` makes me think it's at the beginning of the file which it's not.

Maybe just append this error message with "add a semicolon"?

**Comment by cmoog** [comment]
@kevinburkemeter For which database engine?

**Comment by kevinburkemeter** [comment]
Ah, Postgres

**Comment by kyleconroy** [comment]
We can't provide this information right now due to the lack of structured error messages being returned from pg_query_go https://github.com/lfittl/pg_query_go/issues/24

**Comment by dskarataev** [comment]
yes please! it is very confusing. I would NEVER realize what is the root cause of the error if I would not find this issue on GitHub...

So @kyleconroy would be really good if you would still find some way to add the structured error messages

**Comment by videate-karim** [comment]
pg_query_go was moved. here is the correct link: https://github.com/pganalyze/pg_query_go/issues/24

I looked at this, @kyleconroy - is there any reason we can't track the line number of the source file? In the postgresql `Parse()` function, when `p.Parse()` returns err, couldn't that be emitted and/or returned with the line of source file?

I am looking here: https://github.com/kyleconroy/sqlc/blob/e2c60566caec2ce7276824e46da792cd98fa05c3/internal/engine/postgresql/parse.go#L165

**Comment by kyleconroy** [comment]
> is there any reason we can't track the line number of the source file? 

Yes. The error returned from `pg_query.Parse` does not include a line number nor location. Parse is called once on the contents of a file. This won't be fixed until pg_query_go returns error with this information.

**Comment by StevenACoffman** [comment]
Update: [pg_query_go](https://github.com/pganalyze/pg_query_go) has released this functionality in [v4.2.1](https://github.com/pganalyze/pg_query_go/releases/tag/v4.2.1) and #2288 updates sqlc to use this version

**Comment by andrewmbenton** [comment]
Resolved as of v1.19.0.
